source/Armament.cpp

```diff
@@ -288,3 +360,58 @@ void Armament::Step(const Ship &ship)
 		it.second = max(it.second, 1 - count);
 	}
 }
+
+
+
+void Armament::RecreateViewsAndRanges()
+{
+	turrettedHardpoints.clear();
+	allHardpointsNoAM.clear();
+	fixedHardpoints.clear();
+	antiMissileHardpoints.clear();
+	for(auto &hardpoint : hardpoints)
+	{
+		if(!hardpoint.GetOutfit())
+			continue;
+
+		if(hardpoint.CanAim())
+			turrettedHardpoints.push_back(&hardpoint);
+		else
+			fixedHardpoints.push_back(&hardpoint);
+
+		if(hardpoint.IsAntiMissile())
+			antiMissileHardpoints.push_back(&hardpoint);
+		else
+			allHardpointsNoAM.push_back(&hardpoint);
+	}
+	std::tie(minRange, maxRange, maxTurretsRange) = CalculateRanges();
+}
+
+std::tuple<double, double, double> Armament::CalculateRanges() const
```

Comment:
we dont like big tuples in general, I don't mind it as much for min and max range but here we really dont consider the turrets with the rest so there is no reason to do it all in the same function
how about a CalculateTurretRange() method that does just that?

That said you could also just look at all weapons without AM being considered and affect turret range if its a turret, then I could understand why its all in the same function but here we dont need to do it this way really

I guess you did it this way because the minimal range is the min between the two minimal ranges? That could be calculated in a min(minRange, minRangeTurrets) easily